src/main/java/com/java/stack/StackDemo.java

```diff
@@ -0,0 +1,103 @@
+package com.java.stack;
+
+import java.util.Arrays;
+
+public class StackDemo {
+	private static int CAPACITY = 10;
+	private int size;
+	private int checkfull;
+	private int[] arr = new int[CAPACITY];
+
+	/**
+	 * Add a element to head array
+	 *
+	 */
+	public void add(int item) {
+		if (this.size == arr.length) {
+			// 1. Tao mot mang moi co kich thuoc lon hon gap 1.5 lan
+			int[] newArr = new int[(int) (this.size * 1.5)];
+			// 2. Copy mang cu vao mang moi
+			for (int i = 0; i < arr.length; i++) {
+				newArr[i] = arr[i];
+			}
+			// 3. Gan' item vao mang moi
+			newArr[arr.length] = item;
+			// 4. Gan mang moi vao mang cu
+			this.arr = newArr;
+		} else {
+			arr[this.size] = item;
+		}
+		// 5 Chi tang size mang thi item khac 0
+		if (item != 0) {
+			this.size++;
+		}
+
+	}
+
+	/**
+	 * add an element to array, located by index ( defined by user ) & value (
+	 * defined by user)
+	 *
+	 */
+	public void add(int item, int index) {
+	}
+
+	/**
+	 * remove an element at TOP of array
+	 *
+	 */
+	public void pop() {
+		int remove = 0;
+		// 1. Xác định vị trí đầu tiên có value khác 0 từ phải qua trái của mảng
+		for (int i = arr.length - 1; i > 0; i--) {
+			if (arr[i] != 0) {
+				remove = i;
+				System.out.println("You have just pop => " + arr[i]);
+				break;
+			}
+		}
+		// 2. Khai báo một mảng có kích thướng nhỏ hơn mảng cũ 1 đơn vị
+		int[] newArr = new int[arr.length - 1];
```

Comment:
xảy ra trường hợp, là người dùng add số 0 vào, và họ muốn pop cái số 0 vừa nhập ấy ra!
gơi ý : e có size của stack này, nó xác định được vị trí của phần tử cuối cùng add vào

Reply:
Dạ.

Reply:
i fixed this problem